rcap/http_rulefilter.go

```diff
@@ -0,0 +1,113 @@
+package rcap
+
+import (
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	ErrHttpDisallowed   = errors.New("requests to insecure HTTP endpoints is disallowed")
+	ErrIPsDisallowed    = errors.New("requests to IP addresses are disallowed")
+	ErrDomainDisallowed = errors.New("requests to this domain are disallowed")
+)
+
+// HTTPRules is a set of rules that governs use of the HTTP capability
+type HTTPRules struct {
+	AllowedDomains []string `json:"allowedDomains" yaml:"allowedDomains"`
+	BlockedDomains []string `json:"blockedDomains" yaml:"blockedDomains"`
+	AllowIPs       bool     `json:"allowIPs" yaml:"allowIPs"`
+	AllowHTTP      bool     `json:"allowHTTP" yaml:"allowHTTP"`
+}
```

Comment:
What about non-standard HTTP ports? How could we allow HTTPS-only on :8443, for example?

I could also see a more robust matcherFunc sort of thing where it matches on a regexp instead of just the wildcard. (something like this maybe: https://regex101.com/r/vxiHFH/1)

Reply:
Hmmm, ports isn't something I'd considered. What kind of rule would you want to see there?

I'm specifically avoiding regex here, because they have a history of being fraught with security issues and I'd rather keep it simple.

Reply:
As for ports, we can just treat them as different domains. See suggested changes below.

You're right about regex being scary. If Vendors want more granular or programatic control over what routes can be accessed, they could integrate it into their libraries.

Reply:
It could be also useful if IPs could be specified in CIDR ranges. e.g. 10.0.0.48/29 (10.0.0.48 to 10.0.0.55). IPv6 would be handy as well.

Reply:
Filed as a follow-up here: https://github.com/suborbital/reactr/issues/141